pootle/core/browser.py

```diff
@@ -100,7 +100,12 @@ def make_generic_item(path_obj):
 
 
 def make_directory_item(directory):
-    item = make_generic_item(directory)
+    filters = {}
+
+    if len(VirtualFolder.get_matching_for(directory.pootle_path)):
```

Comment:
As I see this will incur in N*M+1 queries, where:
* N is the number of directories being displayed in the overview,
* M is the number of filtering rules in each browsable `VirtualFolder`, and
* 1 is the query needed for retrieving browsable `VirtualFolder`s

Looks excessive on a first sight, even more taking into account there's no caching of any sort.